python/ray/serve/task_processor.py

```diff
@@ -0,0 +1,254 @@
+
+    @abstractmethod
+    async def get_task_status(self, task_id) -> Dict[str, Any]:
+        pass
+
+    @abstractmethod
+    async def cancel_task(self, task_id) -> bool:
+        pass
+
+    @abstractmethod
+    async def get_metrics(self) -> Dict[str, Any]:
+        pass
+
+    @abstractmethod
+    def start_consumer(self, **kwargs):
+        pass
+
+    @abstractmethod
+    def stop_consumer(self, timeout: float = 10.0):
+        pass
+
+    @abstractmethod
+    def shutdown(self):
+        pass
+
+    @abstractmethod
+    async def health_check(self):
+        pass
+
+
+class CeleryTaskProcessorAdapter(TaskProcessorAdapter):
+    _app: Celery
+    _config: TaskProcessorConfig
+    _worker_thread: Optional[threading.Thread] = None
+
+    def __init__(self, config: TaskProcessorConfig):
+        if not isinstance(config.adapter_config, CeleryTaskProcessorConfig):
+            raise TypeError(
+                "TaskProcessorConfig.adapter_config must be an instance of CeleryTaskProcessorConfig"
+            )
+
+        self._config = config
+
+    def initialize(self, config: TaskProcessorConfig):
+        self._app = Celery(
+            config.queue_name,
+            backend=config.adapter_config.backend_url,
+            broker=config.adapter_config.broker_url,
+        )
+
+        self._app.conf.update(
+            loglevel="info",
+            worker_pool="threads",
+            worker_concurrency=config.adapter_config.worker_concurrency,
+            task_max_retries=config.max_retry,
+            task_default_queue=config.queue_name,
+            task_ignore_result=False,
+            task_acks_late=True,
+            reject_on_worker_lost=True,
+        )
+
+        if config.adapter_config.broker_transport_options is not None:
+            self._app.conf.update(
+                broker_transport_options=config.adapter_config.broker_transport_options,
+            )
+
+        ### TODO(harshit|2025-07-22): add the failed_task_queue_name and unprocessable_task_queue_name business logic here
+
+    def register_task_handle(self, func, name=None):
+        if name:
+            self._app.task(name=name)(func)
+        else:
+            self._app.task(func)
+
+    async def enqueue_task(
+        self, task_name, args=None, kwargs=None, **options
+    ) -> TaskResult:
+        task_response = self._app.send_task(
+            task_name,
+            args=args,
+            kwargs=kwargs,
+            queue=self._config.queue_name,
+            **options,
+        )
+
+        return TaskResult(
+            id=task_response.id,
+            status=task_response.status,
+            created_at=time.time(),
+            result=task_response.result,
+        )
+
+    async def get_task_status(self, task_id) -> Dict[str, Any]:
+        task_details = self._app.AsyncResult(task_id)
+        return {
+            "id": task_details.id,
+            "result": task_details.result,
+            "status": task_details.status,
+        }
```

Comment:
Let's instead return a object of type dataclass here. similar to what you are doing in TaskResult